### PR TITLE
Topic/fusion 73

### PR DESCRIPTION
**Pull request by akamimoor**
## JIRA ticket
[FUSION-73](https://janrain.atlassian.net/browse/FUSION-73)

## Changes

- Refactored the project to make it more testable and understandable
- Added additional logging
- Separated out the AppAuth authentication

Confluence doc for development instructions: [Here](https://janrain.atlassian.net/l/cp/wqc2506v)

[FUSION-73]: https://janrain.atlassian.net/browse/FUSION-73?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Examples/Example-iOS_Swift-Carthage/Source/AppDelegate.swift [hunk 1]
```diff
@@ -1,64 +1,62 @@
...
+    
     func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
-
+        
         if let authorizationFlow = self.currentAuthorizationFlow, authorizationFlow.resumeExternalUserAgentFlow(with: url) {
             self.currentAuthorizationFlow = nil
             return true
         }
-
+
```
Comment (schaeferCode) [inline Examples/Example-iOS_Swift-Carthage/Source/AppDelegate.swift:32]: What's with the new spacing?

**Review by schaeferCode — APPROVED** [review]
Code looks really great! I really like the separation of concerns. App built successfully and everything works as before.

File: Examples/Example-iOS_Swift-Carthage/Source/AppDelegate.swift (hunk 1 above)
Comment (akamimoor) [reply Examples/Example-iOS_Swift-Carthage/Source/AppDelegate.swift:32]: That's weird. The xcode formatter must've added that in. I'll remove it.

File: Examples/Example-iOS_Swift-Carthage/Source/AppDelegate.swift [hunk 2]
```diff
@@ -1,64 +1,62 @@
...
             return true
         }
-
+        
         return false
     }
     
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        // Override point for customization after application launch.
+        return true
```
[the remark this answers is not in the document]
Comment (akamimoor) [reply Examples/Example-iOS_Swift-Carthage/Source/AppDelegate.swift:38]: I could remove that, but it's boilerplate that is typically left in the delegate file.

**Review by ovargasak — APPROVED** [review]
LGTM. objc + swift version build and run.

**Review by kreese-akamai — APPROVED** [review]
:neckbeard: 👍

**Pull request merged**